src/PostRector/Rector/ClassRenamingPostRector.php

```diff
@@ -42,7 +42,12 @@ public function beforeTraverse(array $nodes): array
         foreach ($nodes as $node) {
             if ($node instanceof FileWithoutNamespace || $node instanceof Namespace_) {
                 $removedUses = $this->renamedClassesDataCollector->getOldClasses();
-                $node->stmts = $this->useImportsRemover->removeImportsFromStmts($node->stmts, $removedUses);
+                $stmts = $this->useImportsRemover->removeImportsFromStmts($node->stmts, $removedUses);
+                if ($stmts !== $node->stmts) {
```

Comment:
objects collection can't be compared like this, use something like `$hasChanged` flag instead

Reply:
how would you do it as I can't access the $hasChanged within the UseImportsRemover

Reply:
I found a cleaner solution. Please have a look.